### PR TITLE
Partially enable `D100`, `D102`, `D103` and `D104`

### PR DESCRIPTION
**Pull request by rawwar**
related: #10742 

Enabled D100 to D104 partially by adding ignores to pyproject.toml. 

Also included few fixes which were causing conflictin keys w.r.t `E`* lint issues.

### CODE REVIEW

File: airflow/api/auth/backend/kerberos_auth.py [hunk 1]
```diff
@@ -14,13 +14,9 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+"""Kerberos authentication module."""
 from __future__ import annotations
 
-import warnings
```
Comment (rawwar) [inline airflow/api/auth/backend/kerberos_auth.py:19]: @potiuk , pre-commit hooks reformatted this module after I moved doc string to Line 17 from Line 48. Should I revert this change? 

I thought instead of merging `E`*s and `D`*s , like you mentioned, I could just fix the conflicts. Reason being, I felt `E`*s (at least `E402`) is supposed to be ignored

File: pyproject.toml [hunk 2]
```diff
@@ -1381,83 +1387,1472 @@ combine-as-imports = true
...
+"airflow/www/gunicorn_config.py" = ["D100", "D103"]
+"airflow/www/security.py" = ["D100"]
+"airflow/www/security_appless.py" = ["D100"]
+"airflow/www/security_manager.py" = ["D100", "D102"]
+"airflow/www/session.py" = ["D100"]
+"airflow/www/utils.py" = ["D100", "D102", "D103"]
+"airflow/www/validators.py" = ["D100", "D102"]
+"airflow/www/views.py" = ["D100", "D102", "D103"]
+"airflow/www/widgets.py" = ["D100", "D102"]
+"clients/python/test_python_client.py" = ["D100"]
```
Comment (Taragolis) [inline pyproject.toml]: I think `clients/*` could be excluded from D checks, this one not included into the distribution of airflow or providers

File: airflow/api/auth/backend/kerberos_auth.py (hunk 1 above)
Comment (Taragolis) [reply airflow/api/auth/backend/kerberos_auth.py:19]: @potiuk I guess this license notice might be removed because it is already listed in https://github.com/apache/airflow/blob/main/3rd-party-licenses/LICENSE-flask-kerberos.txt

File: airflow/security/utils.py
Comment (Taragolis) [inline airflow/security/utils.py:1]: And also think it listed in 3rd-party-licenses: https://github.com/apache/airflow/blob/main/3rd-party-licenses/LICENSE-hue.txt

Added in https://github.com/apache/airflow/pull/438

Source
https://github.com/cloudera/hue/blob/master/desktop/core/src/desktop/kt_renewer.py

**Review by dirrao — APPROVED** [review]

File: airflow/security/utils.py
Comment (rawwar) [reply airflow/security/utils.py:1]: So, is it safe to remove from the modules?  If so, I can check for all the modules where E402 is ignored and probably move the license text to a separate license file.

**Comment by potiuk** [comment]
conflicts to solve?

**Comment by rawwar** [comment]
I won't be able to work on this for a week or two. My laptop got damaged. I'll continue working on this once I get back my laptop. 

Thanks!
